File: minwook/2주차/특이한_정렬.java
import java.util.*;

class Solution {
    public int[] solution(int[] numlist, int n) {
        int[] answer = sortArray(numlist, n); // 정렬 함수
        return answer;
    }
    
    public static int[] sortArray(int[] numList, int n) {
        Integer[] integerArray = Arrays.stream(numList).boxed().toArray(Integer[]::new); // int 배열은 Arrays.sort 사용 불가 Integer 배열로 변경
        Arrays.sort(integerArray, Collections.reverseOrder()); // 내림차순 정렬 (절댓값 계산 시 값이 큰게 앞으로 오게함)
        Arrays.sort(integerArray, Comparator.comparingInt(num -> Math.abs(num - n))); // 절댓값 계산 후 정렬
        return Arrays.stream(integerArray).mapToInt(Integer::intValue).toArray(); // int 배열로 변환
    }
}

File: jeonghyeon/2week/특이한 정렬.js
function solution(numlist, n) {
    return numlist.sort((a, b) => {
    const an = Math.abs(a - n); // a와 n의 거리
    const bn = Math.abs(b - n); // b와 n의 거리
    
    if (an == bn) {
      return b - a; // 거리가 같을 때는 큰 숫자를 앞에 오도록 정렬, 양수라면 b가 a보다 큰 값으로 앞으로 정렬
    } else {
      return an - bn; // 거리가 다르면 가까운 거리를 기준으로 정렬, 음수일 경우 an 이 bn 보다 작은 값으로 an이 bn보다 앞으로 정렬
    }
  });
}